src/generic.jl

```diff
@@ -245,8 +281,22 @@ function pairwise!(r::AbstractMatrix, metric::PreMetric, a::AbstractMatrix;
     end
 end
 
+"""
+    pairwise!(r::AbstractMatrix, metric::PreMetric, a, b)
+    pairwise!(r::AbstractMatrix, metric::PreMetric, a)
+
+Compute distances between each element of collection `a` and each element of
+collection `b` according to distance `metric`, and store the result in `r`.
+If a single iterable `a` is provided, compute distances between its elements.
+
+`r` must be a matrix with size `length(a) × length(b)`.
+"""
+pairwise!(r::AbstractMatrix, metric::PreMetric, a, b) = _pairwise!(r, metric, a, b)
+pairwise!(r::AbstractMatrix, metric::PreMetric, a) = _pairwise!(r, metric, a)
+
 """
     pairwise(metric::PreMetric, a::AbstractMatrix, b::AbstractMatrix=a; dims)
+    pairwise(metric::PreMetric, a::AbstractMatrix; dims)
```

Comment:
This is redundant with `b=a` above.
```suggestion
```

Reply:
I don't like this `b=a` thing, because this is not how we do it internally, but that doesn't matter to the user I guess, so it's fine for the docstring.